app/src/main/java/com/spascoding/taskycourse/feature_agenda/presentation/detail_screen/DetailViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package com.spascoding.taskycourse.feature_agenda.presentation.detail_screen
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import javax.inject.Inject
+
+class DetailViewModel @Inject constructor() : ViewModel() {
+
+    var state = MutableStateFlow(DetailViewModelState())
+        private set
+
```

Comment:
I think I commented on that before, but if you use flows, using `private set` doesn't do anything. The StateFlow's value can still be changed from the outside, since the `private set` relates to the flow itself, but not its value. 

As of now, you don't get around two flows:
```
private val _state = MutableStateFlow(DetailViewModelState())
val state = _state.asStateFlow()
```
The private set only works for Compose state + `by` delegate.